HistoricalDataFetcher.Classes/Utilities/ApiRequest.cs

```diff
@@ -0,0 +1,172 @@
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using System.Timers;
+
+namespace HistoricalDataFetcher.Classes.Utilities
+{
+    public static class ApiRequest
+    {
+        private static string _username;
+        private static string _password;
+
+        private static string _httpHost;
+        private static string _urlBase;
+        private static ICache _cache;
+        private static Timer _timer;
+        private static AccessToken _accessTokenInstance;
+
+        public static string HttpHost
+        {
+            get
+            {
+                return _httpHost;
+            }
+        }
+
+        public static string UrlBase
+        {
+            get
+            {
+                return _urlBase;
+            }
+        }
+
+        /// <summary>
+        /// Sets the UrlBase and logs in to test the connection
+        /// </summary>
+        /// <param name="cache">Cache to use</param>
+        /// <param name="username">Username of the Metasys server</param>
+        /// <param name="password">Password of the Metasys server</param>
+        /// <param name="httpHost">Host of the Metasys server</param>
+        /// <returns></returns>
+        public static async Task<bool> InitializeAsync(ICache cache, string username, string password, string httpHost)
+        {
+            _cache = cache;
+            _username = username;
+            _password = password;
+            _httpHost = httpHost;
+            _urlBase = $"https://{_httpHost}/api";
+
+            return await AuthenticateAsync();
+        }
+
+        /// <summary>
+        /// Authenticates and runs the URL with the acceptHeader
+        /// </summary>
+        /// <param name="url">Complete URL</param>
+        /// <param name="acceptHeader">header used to make the call</param>
+        /// <returns>Endpoint Result</returns>
+        public static async Task<string> RunEndpointCallAsync(string url, string acceptHeader)
+        {
+            if (_accessTokenInstance == null)
+            {
+                await AuthenticateAsync();
+            }
+
+            var result = _cache.Get(url);
+            return !string.IsNullOrEmpty(result) ? result : await RunEndpointHelperAsync(url, acceptHeader);
+        }
+
+        /// <summary>
+        /// Runs the URL using the required header info
+        /// </summary>
+        /// <param name="url">Complete URL</param>
+        /// <param name="acceptHeader">header used to make the call</param>
+        /// <returns>EndPoint result</returns>
+        private static async Task<string> RunEndpointHelperAsync(string url, string acceptHeader)
+        {
+            string rawJsonToReturn = string.Empty;
+            var uri = new Uri(url);
+
+            using (var handler = new HttpClientHandler
+            {
+                ServerCertificateCustomValidationCallback = (sender, certificate, chain, sslPolicyErrors) => true,
```

Comment:
No, don't tell customers to do this.

Reply:
This has been addressed via the HandleCertificateError method